src/lib/db.js

```diff
@@ -1578,3 +1442,108 @@ export function getSemesterTeams(semesterName, teamIndices) {
     })
   ));
 }
+
+export async function getNumArticles() {
+  const rows = await database('articles').count('* as numArticles');
+  return rows[0].numArticles;
+}
+
+/**
+ * Updates which category an article belongs to
+ * @param {Object} jsonGraphArg - An object of type { [slugs]: articleUpdateObject[] }
+ * @returns {Promise<string[]>} - Slugs of the articles that had their category changed
+ */
+async function updateArticleCategories(jsonGraphArg) {
+  const articleCategoryPairs = _.map(jsonGraphArg, (articleUpdater, slug) => {
+    if (articleUpdater.hasOwnProperty('category')) {
+      return {
+        articleSlug: slug,
+        categorySlug: articleUpdater.category,
+      };
+    }
+    return null;
+  }).filter(x => x);
+  if (articleCategoryPairs.length === 0) {
+    return [];
+  }
+
+  const categorySlugs = _.uniq(articleCategoryPairs.map(x => x.categorySlug));
+  // Get the ids of each of the categories as that's what we'll have to update the foreign key with
+  const categoryRows = await database.select('slug', 'id').from('categories')
+    .whereIn('slug', categorySlugs);
+
+  const categorySlugToID = {};
+
+  categoryRows.forEach(row => {
+    categorySlugToID[row.slug] = row.id;
+  });
+
+  const updatePromises = articleCategoryPairs.map(pair => {
+    const categoryID = categorySlugToID[pair.categorySlug];
+    const updateObject = { category_id: categoryID };
+    return database('articles').where('slug', '=', pair.articleSlug)
+      .update(updateObject);
+  });
+  // eslint-disable-next-line
+  await Promise.all(updatePromises);
+  return articleCategoryPairs.map(x => x.articleSlug);
+}
+
+/**
+ * Updates articles directly tied to an article
+ * @param {Object} jsonGraphArg - An object of type { [slugs]: articleUpdateObject[] }
+ * @returns {Promise<boolean>} - Whether the update was a success
+ */
+export async function updateArticles(jsonGraphArg) {
+  const updatePromises = _.map(jsonGraphArg, (articleUpdater, slug) => {
+    const processedArticleUpdater = {
+      ...articleUpdater,
+    };
+    // We only store a foreign key to the categories table so we won't update it in this query
+    delete processedArticleUpdater.category;
+    return database('articles').where('slug', '=', slug)
+      .update(processedArticleUpdater);
+  });
+  updatePromises.push(updateArticleCategories(jsonGraphArg));
+  const returnValues = await Promise.all(updatePromises);
+  const articlesWithChangedCategory = returnValues[returnValues.length - 1];
+  // If categories changed make sure issue data is still consistent
+  if (articlesWithChangedCategory.length > 0) {
+    const issueRows = await database.distinct('issue_id').select()
+      .from('issues_posts_order')
```

Comment:
Emil, where you thinking of renaming `issues_posts_order` to `issues_articles_order` for consistency?

Reply:
Yes! It is part of what I wrote in the initial description of this PR:

> Rename some more tables so for example nothing uses the posts terminology anymore and make everything reference articles

Reply:
But as I mentioned there, I didn't want to include more renaming and other things non-crucial for making things work in this PR, I'll follow up in separate PRs with tests and renamings